ResearchKit/Common/ORKNavigableOrderedTask.m

```diff
@@ -74,6 +76,42 @@ - (void)removeNavigationRuleForTriggerStepIdentifier:(NSString *)triggerStepIden
     [_stepNavigationRules removeObjectForKey:triggerStepIdentifier];
 }
 
+- (NSDictionary<NSString *, ORKStepNavigationRule *> *)stepNavigationRules {
+    if (!_stepNavigationRules) {
+        return (NSDictionary<NSString *, ORKStepNavigationRule *> *)@[];
+    }
+    return [_stepNavigationRules copy];
+}
+
+- (void)setSkipNavigationRule:(ORKSkipStepNavigationRule *)skipStepNavigationRule forStepIdentifier:(NSString *)stepIdentifier {
+    ORKThrowInvalidArgumentExceptionIfNil(skipStepNavigationRule);
+    ORKThrowInvalidArgumentExceptionIfNil(stepIdentifier);
+    
+    if (!_skipStepNavigationRules) {
+        _skipStepNavigationRules = [NSMutableDictionary new];
+    }
+    _skipStepNavigationRules[stepIdentifier] = skipStepNavigationRule;
+}
+
+- (ORKSkipStepNavigationRule *)skipNavigationRuleForStepIdentifier:(NSString *)stepIdentifier {
+    ORKThrowInvalidArgumentExceptionIfNil(stepIdentifier);
+    
+    return _skipStepNavigationRules[stepIdentifier];
+}
+
+- (void)removeSkipNavigationRuleForStepIdentifier:(NSString *)stepIdentifier {
+    ORKThrowInvalidArgumentExceptionIfNil(stepIdentifier);
+    
+    [_skipStepNavigationRules removeObjectForKey:stepIdentifier];
+}
+
+- (NSDictionary<NSString *, ORKSkipStepNavigationRule *> *)skipStepNavigationRules {
+    if (!_skipStepNavigationRules) {
+        return (NSDictionary<NSString *, ORKSkipStepNavigationRule *> *)@[];
```

Comment:
Fixed! By the way, I've noticed that if you `return @[]` from within these methods, even without the dictionary cast, *Xcode 7 Version 7.3.1 (7D1014)* doesn't produce a warning. It seems it was a bug, *Xcode 8 beta* does produce the warning. ;-)